client/rest-high-level/src/main/java/org/elasticsearch/client/SecurityClient.java

```diff
@@ -0,0 +1,91 @@
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.protocol.xpack.security.PutUserRequest;
+import org.elasticsearch.protocol.xpack.security.PutUserResponse;
+
+import java.io.IOException;
+
+import static java.util.Collections.emptySet;
+
+/**
+ * A wrapper for the {@link RestHighLevelClient} that provides methods for accessing the Security APIs.
+ * <p>
+ * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api.html">Security APIs on elastic.co</a>
+ */
+public final class SecurityClient {
+
+    private final RestHighLevelClient restHighLevelClient;
+
+    SecurityClient(RestHighLevelClient restHighLevelClient) {
+        this.restHighLevelClient = restHighLevelClient;
+    }
+
+    /**
+     * Create/update a user in the native realm synchronously.
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-users.html">
+     * the docs</a> for more.
+     * @param request the request with the user's information
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the response from the put user call
+     * @throws IOException in case there is a problem sending the request or parsing back the response
+     */
+    public PutUserResponse putUser(PutUserRequest request, RequestOptions options) throws IOException {
+        return restHighLevelClient.performRequestAndParseEntity(request, RequestConverters::putUser, options,
+            SecurityClient::parsePutUserResponse, emptySet());
+    }
+
+    /**
+     * Asynchronously create/update a user in the native realm.
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-users.html">
+     * the docs</a> for more.
+     * @param request the request with the user's information
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @param listener the listener to be notified upon request completion
+     */
+    public void putUserAsync(PutUserRequest request, RequestOptions options, ActionListener<PutUserResponse> listener) {
+        restHighLevelClient.performRequestAsyncAndParseEntity(request, RequestConverters::putUser, options,
+            SecurityClient::parsePutUserResponse, listener, emptySet());
+    }
+
+    /**
+     * Parses the rest response from the put user API. The rest API wraps the XContent of the
+     * {@link PutUserResponse} in a field, so this method unwraps this value.
+     *
+     * @param parser the XContent parser for the response body
+     * @return the {@link PutUserResponse} parsed
+     * @throws IOException in case there is a problem parsing the response
+     */
+    private static PutUserResponse parsePutUserResponse(XContentParser parser) throws IOException {
```

Comment:
yuck. Is it possible to break this in the API for 7.0? Do we want to keep that custom wrapping in `RestPutUserAction`? Is it necessary?

Reply:
In this PR I have added a solution that prevents API breakage in 6.x by keeping the wrapping AND having the value unwrapped. For 7.0 we can remove the wrapped value. Does that work for you?